chemprop/models/mpn.py

```diff
@@ -265,22 +311,22 @@ def forward(self,
             if self.features_only:
                 return features_batch
 
-        if self.atom_descriptors == 'descriptor':
+        if self.atom_descriptors == 'descriptor' or self.bond_descriptors == 'descriptor':
             if len(batch) > 1:
                 raise NotImplementedError('Atom descriptors are currently only supported with one molecule '
                                           'per input (i.e., number_of_molecules = 1).')
 
-            encodings = [enc(ba, atom_descriptors_batch) for enc, ba in zip(self.encoder, batch)]
+            encodings = [enc(ba, atom_descriptors_batch, bond_descriptors_batch) for enc, ba in zip(self.encoder, batch)]
         else:
             if not self.reaction_solvent:
-                 encodings = [enc(ba) for enc, ba in zip(self.encoder, batch)]
+                encodings = [enc(ba) for enc, ba in zip(self.encoder, batch)]
             else:
-                 encodings = []
-                 for ba in batch:
-                     if ba.is_reaction:
-                         encodings.append(self.encoder(ba))
-                     else:
-                         encodings.append(self.encoder_solvent(ba))
+                encodings = []
+                for ba in batch:
+                    if ba.is_reaction:
+                        encodings.append(self.encoder(ba))
+                    else:
+                        encodings.append(self.encoder_solvent(ba))
 
         output = reduce(lambda x, y: torch.cat((x, y), dim=1), encodings)
```

Comment:
I know this line isn't your doing, but isn't this just doing the following:
```python
xs: Iterable[Tensor] = [x1, x2, x3]
X = torch.cat((torch.cat((x1, x2), dim=1), x3), dim=1)
```
I.e., progressively concatenating the next tensor onto the result of concatenating the previous two tensors? If so, why not just call `torch.cat(encodings, dim=1)`

Reply:
I thought that is progressively concatenating. I have modified the code.